test/e2e/node/node_problem_detector.go

```diff
@@ -103,38 +75,40 @@ var _ = SIGDescribe("NodeProblemDetector", func() {
 		// KubeletStart event since there is no easy way to check when test has actually started.
 		checkForKubeletStart := false
 
-		for _, host := range hosts {
-			cpuUsageStats[host] = []float64{}
-			uptimeStats[host] = []float64{}
-			rssStats[host] = []float64{}
-			workingSetStats[host] = []float64{}
+		for _, node := range nodes {
+			cpuUsageStats[node.Name] = []float64{}
+			uptimeStats[node.Name] = []float64{}
+			rssStats[node.Name] = []float64{}
+			workingSetStats[node.Name] = []float64{}
 
 			cmd := "systemctl status node-problem-detector.service"
-			result, err := e2essh.SSH(ctx, cmd, host, framework.TestContext.Provider)
-			isStandaloneMode[host] = (err == nil && result.Code == 0)
+			result, err := hostExec.Execute(ctx, cmd, &node)
+			framework.Logf("output from cmd on node %s, value:%+v", node.Name, result)
```

Comment:
This is why the test is failing

```
Oct 12 10:52:25.297: INFO: ExecWithOptions: execute(POST https://34.139.220.225/api/v1/namespaces/node-problem-detector-9548/pods/hostexec-gce-cluster-minion-group-vw9f-vjdc4/exec?command=nsenter&command=--mount%3D%2Frootfs%2Fproc%2F1%2Fns%2Fmnt&command=--&command=sh&command=-c&command=systemctl+status+node-problem-detector.service&container=agnhost-container&container=agnhost-container&stderr=true&stdout=true)
Oct 12 10:52:25.612: INFO: output from cmd on node gce-cluster-minion-group-vw9f, value:{Host:gce-cluster-minion-group-vw9f Cmd:systemctl status node-problem-detector.service Stdout: Stderr:Failed to connect to bus: No data available
 Code:1}
STEP: Inject log to trigger DockerHung on node "gce-cluster-minion-group-vw9f" - test/e2e/node/node_problem_detector.go:129 @ 10/12/23 10:52:25.612
Oct 12 10:52:25.612: INFO: ExecWithOptions {Command:[nsenter --mount=/rootfs/proc/1/ns/mnt -- sh -c sudo sh -c "echo 'kernel: INFO: task docker:12345 blocked for more than 120 seconds.' >> /dev/kmsg"] Namespace:node-problem-detector-9548 PodName:hostexec-gce-cluster-minion-group-vw9f-vjdc4 ContainerName:agnhost-container Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Oct 12 10:52:25.612: INFO: >>> kubeConfig: /workspace/.kube/config
Oct 12 10:52:25.613: INFO: ExecWithOptions: Clientset creation
Oct 12 10:52:25.613: INFO: ExecWithOptions: execute(POST https://34.139.220.225/api/v1/namespaces/node-problem-detector-9548/pods/hostexec-gce-cluster-minion-group-vw9f-vjdc4/exec?command=nsenter&command=--mount%3D%2Frootfs%2Fproc%2F1%2Fns%2Fmnt&command=--&command=sh&command=-c&command=sudo+sh+-c+%22echo+%27kernel%3A+INFO%3A+task+docker%3A12345+blocked+for+more+than+120+seconds.%27+%3E%3E+%2Fdev%2Fkmsg%22&container=agnhost-container&container=agnhost-container&stderr=true&stdout=true)
Oct 12 10:52:28.032: INFO: ExecWithOptions
```
hostExec for systemctl is failing. the `isStandslonemode` branch won't run if this command fails and it will try to check for pods we didn't deploy

On COS, we deploy npd via standalone instead of using daemonset

https://github.com/kubernetes/kubernetes/blob/a2cc9db02fa6420c7905ef372cae6695fddb1120/cluster/gce/config-default.sh#L290-L295

pull-kubernetes-e2e-gce is successful because it runs on ubuntu and NPD is deployed as a daemonset.

I can dig around the hostExec pod business logic or change cos to use daemonsets and break the kubelet standalone tests(this is an alpha feature so I don't mind breaking it and fixing it later). NPD tests are already broken for other reasons.

https://testgrid.k8s.io/sig-node-node-problem-detector#ci-npd-e2e-kubernetes-gce-gci